File: Práctica2/LineaCirculo.py
import math
class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    def coord_cartesianas(self):
        return self.x, self.y
    def coord_polares(self):
        radio = math.sqrt(self.x * self.x + self.y * self.y)
        angulo = math.atan(self.y / self.y)
        angulo = math.degrees(angulo)
        return radio, angulo
    def __str__(self):
        return "({:.2f},{:.2f})".format(self.x, self.y)

class Linea:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f"Línea de {self.p1} a {self.p2}"

    def dibujaLinea(self):
        print(f"Dibujando línea desde {self.p1} hasta {self.p2}")

# Definición de la clase Circulo
class Circulo:
    def __init__(self, centro, radio):
        self.centro = centro
        self.radio = radio

    def __str__(self):
        return f"Círculo con centro en {self.centro} y radio {self.radio}"

    def dibujaCirculo(self):
        print(f"Dibujando un círculo con centro en {self.centro} y radio {self.radio}")
p1 = Punto(0,3)
print(p1)
x,y = p1.coord_cartesianas()
print(x,y)
r,a = p1.coord_polares()
print(r,a) 

p2 = Punto(4, 0)

l1 = Linea(p1, p2)
print(l1)
l1.dibujaLinea()

c1 = Circulo(p1, 5)
print(c1)
c1.dibujaCirculo()

File: Práctica2/LineaCirculo.java
package punto;

public class Punto {
    public float x;
    public float y;
    
    public Punto(float x, float y)
    {
        this.x = x;
        this.y = y;
    }
    
    public float[] coord_cartesianas()
    {
        return new float[] {this.x, this.y};
    }
    
    public float[] coord_polares()
    {
        float radio = (float)Math.sqrt(this.x * this.x + this.y * this.y);
        float angulo = (float)Math.sqrt(this.x/this.y);
        angulo = (float)Math.toDegrees(angulo);
        return new float[] {radio, angulo};
    }
    
    @Override
    public String toString()
    {
        return String.format("(%.2f, %.2f)", this.x, this.y);
    }
    
    public static void main(String[] args) {
        // TODO code application logic here
        Punto p1 = new Punto(0, 3);
        System.out.println("-------------Punto p1----------------");
        float a[] = p1.coord_cartesianas();
        System.out.println(a[0]+" "+a[1]);
        float b[] = p1.coord_polares();
        System.out.println(b[0]+" "+b[1]);
        
        Punto p2 = new Punto(0, 3);
        System.out.println("-------------Punto p2----------------");
        float c[] = p2.coord_cartesianas();
        System.out.println(c[0]+" "+c[1]);
        float d[] = p2.coord_polares();
        System.out.println(d[0]+" "+d[1]);
        
        Linea l1 = new Linea(p2, p1);
        System.out.println("---------------------Linea------------------");
        l1.dibujaLinea();
        
        Circulo c1 = new Circulo(p2, 4);
        System.out.println("---------------------Circulo----------------");
        c1.dibujaCirculo();
    }   
}

class Linea {
    private Punto p1;
    private Punto p2;

    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    @Override
    public String toString() {
        return "Línea de " + p1 + " a " + p2;
    }

    public void dibujaLinea() {
        System.out.println("Dibujando linea desde " + p1 + " hasta " + p2);
    }
}

class Circulo{
    public Punto centro;
    public float radio;
    
    public Circulo(Punto c, float r)
    {
        this.centro = c;
        this.radio = r;
    }
    
    @Override
    public String toString()
    {
        return "Circulo con centro en "+centro+" y radio "+radio;
    }
    
    public void dibujaCirculo()
    {
        System.out.println("Dibujando un circulo con centro en "+centro+" y radio "+radio);
    } 
}